static/js/index.js

```diff
@@ -0,0 +1,133 @@
+const API_PREFIX = 'api';
+const baseUrl = window.location.origin;
+
+const setScrapingLogs = (logs) => {
+  const logsNode = document.getElementById('scraping-logs');
+  logs.forEach((el) => {
+    const pNode = document.createElement('p');
+    pNode.innerText = el;
+    logsNode.appendChild(pNode);
+  });
+};
+
+const setRadioButtons = (scraperSettings) => {
+  document.querySelectorAll(`[name="loop-radio-group"][value="${scraperSettings.loop_mode.toString()}"]`)[0].checked = true;
+  document.querySelectorAll(`input[name="media-radio-group"][value="${scraperSettings.media_mode}"]`)[0].checked = true;
+  document.querySelectorAll(`input[name="ids-radio-group"][value="${scraperSettings.ids_source}"]`)[0].checked = true;
+};
+
+const checkShutdown = async () => {
+  const makeStatusRequest = async () => {
+    const requestUrl = `${baseUrl}/${API_PREFIX}/scraper/status/`;
+    return await (await fetch(requestUrl)).json();
+  };
+
+  const checkStatusEveryMs = 2000;
+  const updatePage = async () => {
+    try {
+      const response = await makeStatusRequest();
+      if (response.status === 'stopped') {
+        location.reload();
+      } else {
+        setTimeout(updatePage, checkStatusEveryMs);
+      }
+    } catch {
+      location.reload();
+    }
+  };
+  setTimeout(updatePage, checkStatusEveryMs);
+};
+
+const updateCommandButton = (buttonStatus) => {
+  const commandButton = document.getElementById('command-button');
+  if (buttonStatus === 'running') {
+    commandButton.disabled = false;
+    commandButton.classList = 'btn stop-btn';
+    commandButton.innerText = 'STOP';
+    commandButton.value = 'stop';
+  } else if (buttonStatus === 'shutdown') {
+    commandButton.disabled = true;
+    commandButton.classList = 'btn update-btn';
+    commandButton.innerText = 'updating...';
+    checkShutdown();
+  } else {
+    commandButton.disabled = false;
+    commandButton.classList = 'btn start-btn';
+    commandButton.innerText = 'START';
+    commandButton.value = 'start';
+  }
+};
+
+const getScraperStatus = async () => {
+  const statusUrl = `${baseUrl}/${API_PREFIX}/scraper/status/`;
+  const settingsUrl = `${baseUrl}/${API_PREFIX}/scraper/settings/`;
+  const statusResponseData = await (await fetch(statusUrl)).json();
+  const settingsResponseData = await (await fetch(settingsUrl)).json();
+  const outputNode = document.getElementById('scraping-results');
+  setScrapingLogs(statusResponseData.log_lines);
+  setRadioButtons(settingsResponseData);
+  updateCommandButton(statusResponseData.status);
```

Comment:
nit: move this one up so it's next to setScrapingLogs as it uses the same response